### PR TITLE
fix(input): input maxlength calculate the correct length

### PR DESCRIPTION
**Pull request by wen-haoming**
Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by wen-haoming** [comment]
fix https://github.com/element-plus/element-plus/issues/2653

File: packages/input/src/index.vue
Comment (JeremyWuuuuu) [inline packages/input/src/index.vue:349]: Add a space after `if`

Comment (JeremyWuuuuu) [reply packages/input/src/index.vue:349]:
```suggestion
If (upperLimit.value) {
```

File: packages/input/src/index.vue
Comment (JeremyWuuuuu) [inline packages/input/src/index.vue:259]: Since we have `props.maxlength` why bother creating a computed which only returns `props.maxlength` anyway?

File: packages/input/src/index.vue
Comment (JeremyWuuuuu) [inline packages/input/src/index.vue:257]: Here we are creating this `computed` is because `attrs` is not reactive.

File: website/docs/zh-CN/input.md
Comment (JeremyWuuuuu) [inline website/docs/zh-CN/input.md:954]: Reminder:
You need to update all language documentations, if the documentation’s language is not your familiar language, please use translator to translate it direct to English, for all languages.

**Comment by JeremyWuuuuu** [comment]
I’ve left few comments, please address them before it gets merged, I am happy to approve this PR after all comments are resolved. Thank you for your contribution! 👍

**Comment by wen-haoming** [comment]
> I’ve left few comments, please address them before it gets merged, I am happy to approve this PR after all comments are resolved. Thank you for your contribution! 👍

It's fixed ~

File: packages/input/src/index.vue
Comment (wen-haoming) [reply packages/input/src/index.vue]: Got it.

File: packages/input/src/index.vue
Comment (JeremyWuuuuu) [inline packages/input/src/index.vue:286]: I found a potential bug though, the type of `maxlength` says it can be both `string` and `number` type, I think yo need to parse `maxlength` before comparison operation. WDYT?

File: packages/input/src/index.vue
Comment (JeremyWuuuuu) [inline packages/input/src/index.vue:351]: Like here you used `Number(index)` to make sure that the value is `number` even though `'5' > '4'` holds `true` but that doesn't mean that we can do so.

File: packages/input/src/index.vue
Comment (wen-haoming) [reply packages/input/src/index.vue:286]: It is my opinion that 

- The maxlength only allows the number type  
```js
    maxlength: {
      type: Number,
    },
```
- `return isWordLimitVisible.value && (textLength.value > Number(props.maxlength) )`  Compatible with past string types

File: packages/input/src/index.vue
Comment (JeremyWuuuuu) [inline packages/input/src/index.vue:225]: @wen-haoming but you made the type `string` acceptable here, so I thought you need to do a string parse before doing actual arithmetic operations

Comment (JeremyWuuuuu) [reply packages/input/src/index.vue:225]: Since we do not want to break the old functionality as passing that to `<input />` which accepts both `string` and `number` type. I do think that you need to perform a `Number` or `parseInt` for `maxlength` though.

Comment (wen-haoming) [reply packages/input/src/index.vue:225]: I think so 😄

**Review by jw-foss — APPROVED** [review]

**Pull request merged**
